src/olympia/activity/tasks.py

```diff
@@ -13,14 +13,21 @@
 @write
 def process_email(message, **kwargs):
     """Parse emails and save activity log entry."""
-    msg_id = message.get('MessageId', None)
+    # Some emails (gmail, at least) come with Message-ID instead of MessageId.
+    msg_id = message.get('MessageId')
+    if not msg_id:
+        custom_headers = message.get('CustomHeaders', [])
```

Comment:
the message parameter is `.get('Message')`[[1](https://github.com/eviljeff/olympia/blob/f2dafe975078d7d811e28749867a3ac5570b460d/src/olympia/activity/views.py#L116)] from a payload like https://www.socketlabs.com/api-reference/inbound-api/#json